src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symbol.java

```diff
@@ -426,6 +418,29 @@ public boolean isValue() {
         return (flags() & VALUE) != 0;
     }
 
+    /**
+     * Is this a *derived* reference projection symbol ??
+     */
+    public boolean isReferenceProjection() {
+        return false;
+    }
+
+    /**
+     * Return the value projection IFF `this' happens to be derived reference projection, null
+     * otherwise.
+     */
+    public Symbol valueProjection() {
+        return null;
+    }
+
+    /**
+     * Return the reference projection IFF `this' happens to be value projection, null
+     * otherwise.
+     */
+    public Symbol referenceProjection() {
+        return null;
```

Comment:
How much would it help to have this return `this` ? It seems to me that there a lot of `if <value> ... referenceProjection()` going around.

Reply:
It could result in cleaner code. I have this one JDK-8244232 to cover it.